### ISSUE TITLE
properly escape urls

### ISSUE THREAD

**Issue by chdsbd** [chdsbd/vscode-githubinator#28]
`https://github.com/sublimehq/Packages/blob/master/C#/C#.sublime-syntax` should be `https://github.com/sublimehq/Packages/blob/master/C%23/C%23.sublime-syntax`

```
test#? .,<>@!#$%^^&*()_+=":;?.,~`"'[{]}\|
=> 
test%23%3F%20.%2C%3C%3E%40!%23%24%25%5E%5E%26*()_%2B%3D%22:%3B%3F.%2C%7E%60%22'%5B%7B%5D%7D%5C%7C
```
related: https://github.com/ehamiter/GitHubinator/issues/53

__EDIT:__ I wanted to use `vscode.Uri` everywhere to fix this, but it seems that calling `with({query: ...})` incorrectly applies encodeURIComponent on the string, which breaks the Gitlab provider for opening a URI. This issue could be fixed, but it's kind of a pain I don't think I'll touch it until I personally run into this issue.

**Comment by chdsbd** [comment]
fixed in #29

**Comment by chdsbd** [comment]
This wasn't fixed 😞

**Comment by chdsbd** [comment]
related: https://github.com/microsoft/vscode/issues/85930

**Comment by chdsbd** [comment]
Fixed in #40
